app/jobs/process_delius_data_job.rb

```diff
@@ -84,20 +87,24 @@ def process_record(probation_record, nomis_offender_id, trigger_method)
           tags: tags,
           system_event: true,
           data: {
-            'before' => case_info_atts_before,
-            'after' => case_information_after.attributes.except(*atts_to_ignore)
+            'before' => case_info_attrs_before,
+            'after' => case_info.slice(case_info_attrs_before.keys)
           }
         )
       else
-        case_information_after.errors.each do |error|
+        case_info.errors.each do |error|
           DeliusImportError.create! nomis_offender_id: nomis_offender_id,
                                     error_type: error_type(error.attribute)
         end
       end
+    else
+      logger.info(
+        "nomis_offender_id=#{nomis_offender_id},trigger_method=#{trigger_method},job=process_delius_data_job,event=case_information_unchanged"
+      )
     end
   end
 
-  def map_delius_to_case_info(probation_record, case_info)
+  def map_delius_to_case_info!(probation_record, case_info)
```

Comment:
Just to confirm, `tap` called in here is not idempotent right? Meaning that the supplied case_info object will actually be altered by this function, not just the return value?

Reply:
Yep correct, it is intended to be modifying the passed `case_info`. It was always like that but the previous code was a bit misleading, creating a second variable that is not needed. I've added the bang at the end of the method name to be explicit about the in-place modifications.